drivers/clock_control/clock_control_rpi_pico.c

```diff
@@ -141,7 +158,9 @@ enum rpi_pico_clkid {
 	rpi_pico_clkid_clk_peri = RPI_PICO_CLKID_CLK_PERI,
 	rpi_pico_clkid_clk_usb = RPI_PICO_CLKID_CLK_USB,
 	rpi_pico_clkid_clk_adc = RPI_PICO_CLKID_CLK_ADC,
+#if defined(RPI_PICO_CLKID_CLK_RTC)
```

Comment:
Please define and use a macro like the one below at the head.
The condition should check the existence of hardware via the devicetree.
Also others.

```
#define CLK_EXISTS(clk)                                                                            \
       DT_NODE_HAS_STATUS(DT_INST_CLOCKS_CTLR_BY_NAME(0, clk), okay)
```